components/ws-manager-mk2/controllers/timeout_controller.go

```diff
@@ -202,6 +202,19 @@ func (r *TimeoutReconciler) isWorkspaceTimedOut(ws *workspacev1.Workspace) (reas
 			}
 			return decide(start, timeout, activityNone)
 		} else if isClosed {
+			reason := func() string {
+				afterClosed := timeouts.AfterClose
+				if customClosedTimeout := ws.Spec.Timeout.ClosedTimeout; customClosedTimeout != nil {
+					afterClosed = util.Duration(customClosedTimeout.Duration)
+					if afterClosed == 0 {
+						return ""
+					}
+				}
+				return decide(*lastActivity, afterClosed, activityClosed)
+			}()
+			if reason != "" {
+				return reason
+			}
 			return decide(*lastActivity, timeouts.AfterClose, activityClosed)
```

Comment:
should this line be removed now? It's no longer there in mk1

Reply:
Ah that would make sense. It couldn't have been the `omitempty` what I said below, as then it also the `0s` wouldn't show up in the CRD, but it was there